### PR TITLE
feat(design-system): add inline message component and stories, fix banner component icon

### LINKED ISSUE #2017: [React] Add InlineMessage component
**Time estimation: 0.5d**

### PR DESCRIPTION
**Pull request by tshuli**
**Features**:

- add inline message component and stories. closes #2017 
- also fixes minor issue with banner component not displaying correct icons

## Before & After Screenshots
In storybook

### CODE REVIEW

File: frontend/src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -50,7 +51,10 @@ export const Banner = ({
       <Box __css={styles.banner}>
         <Flex sx={styles.item}>
           <Flex>
-            <Icon as={BxsInfoCircle} __css={styles.icon} />
+            <Icon
+              as={variant === 'info' ? BxsInfoCircle : BxsErrorCircle}
```
Comment (karrui) [inline frontend/src/components/Banner/Banner.tsx:55]: Thanks for the catch!

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  )
+
+  return (
+    <Box __css={styles.messagebox}>
+      <Flex sx={styles.item}>
+        <Icon
+          as={variant !== 'error' ? BxsInfoCircle : BxsErrorCircle}
+          __css={styles.icon}
+        />
+        <ReactMarkdown components={mdComponents}>{children}</ReactMarkdown>
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:38]: I realized we are forcing people to use this component with markdown. Should we make it an option instead, so users of this component can pass in anything they want, and only use markdown parser if they select that option?

Should change it for the `Banner` component too (was tunnel visioned on the way us FormSG uses banner currently, but not for all use cases)

Should we add a new prop `useMarkdown?: boolean` and only wrap with `ReactMarkdown` if that is `true`? This way, users can do this:

```tsx
return (
  <InlineMessage>
    <Box>Put anything here, does not have to be markdown</Box>
    <Input />
  </InlineMessage>
)

// Or this
return <InlineMessage useMarkdown>Now **this** is _markdown_</InlineMessage>
```

File: frontend/src/components/InlineMessage/InlineMessage.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Meta, Story } from '@storybook/react'
+
+import { InlineMessage, InlineMessageProps } from './InlineMessage'
+
+export default {
+  title: 'Components/InlineMessage',
+  component: InlineMessage,
+  argTypes: {
+    variant: {
+      options: ['info', 'warning', 'error'],
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.stories.tsx:10]: does it not autoconvert the argTypes with the exported `InlineMessageVariant` type? (may need to restart storybook so the builder rebuilds the app)

File: frontend/src/components/Banner/Banner.tsx [hunk 4]
```diff
@@ -50,7 +51,10 @@ export const Banner = ({
...
+              __css={styles.icon}
+            />
             <ReactMarkdown components={mdComponents}>{children}</ReactMarkdown>
```
Comment (karrui) [inline frontend/src/components/Banner/Banner.tsx:58]: As stated, should change this component to accept any children, or only markdown with `useMarkdown`. Do you want to do it or should I do it in a separate PR?

File: frontend/src/components/InlineMessage/InlineMessage.tsx (hunk 2 above)
Comment (tshuli) [reply frontend/src/components/InlineMessage/InlineMessage.tsx]: thanks! added `useMarkdown`, see if implementation is ok

File: frontend/src/components/InlineMessage/InlineMessage.stories.tsx (hunk 3 above)
Comment (tshuli) [reply frontend/src/components/InlineMessage/InlineMessage.stories.tsx]: it does actually, made a typing mistake, my bad. corrected

File: frontend/src/components/Banner/Banner.tsx (hunk 4 above)
Comment (tshuli) [reply frontend/src/components/Banner/Banner.tsx]: edited, same as above

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        return <Link {...props} />
+      },
+    }),
+    [],
+  )
+
+  const messageComponent = useMarkdown ? (
+    <ReactMarkdown components={mdComponents}>{children}</ReactMarkdown>
+  ) : (
+    <Box>{children}</Box>
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:36]: should we return `children` directly without wrapping it in a `Box`?

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  const mdComponents: Components = useMemo(
+    () => ({
+      a: (props) => {
+        return <Link {...props} />
+      },
+    }),
+    [],
+  )
+
+  const messageComponent = useMarkdown ? (
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:33]: maybe we can inline this directly?

File: frontend/src/components/InlineMessage/InlineMessage.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+const InlineMessageTemplate: Story<InlineMessageProps> = (args) => (
+  <InlineMessage {...args} />
+)
+
+export const Default = InlineMessageTemplate.bind({})
+Default.args = {
+  variant: 'info',
+  children: `**Markdown** accepted, or just use a normal string.`,
+  useMarkdown: true,
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.stories.tsx:18]: default markdown is false though, so maybe default can showcase the alternative usecase of rendering random things, then add a new `WithMarkdown` story that shows this current default story

File: frontend/src/components/Banner/Banner.stories.tsx [hunk 8]
```diff
@@ -12,25 +12,29 @@ const Template: Story<BannerProps> = (args) => <Banner {...args} />
 export const Default = Template.bind({})
 Default.args = {
   children: `Random string, _can_ be **markdown** too!`,
+  useMarkdown: true,
```
Comment (karrui) [inline frontend/src/components/Banner/Banner.stories.tsx:15]: same here, sorry for the extra work in this PR! :P.

**Review by karrui — COMMENTED** [review]
A couple more comments and we're set! Remember to update the PR description and title so we can easily search for this

**Comment by tshuli** [comment]
thanks @karrui ! changes made for review

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  const styles = useMultiStyleConfig('InlineMessage', { variant })
+
+  const mdComponents: Components = useMemo(
+    () => ({
+      a: (props) => {
+        return <Link {...props} />
+      },
+    }),
+    [],
+  )
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:31]: Seems slightly different from the banner one (banner has external link rendering), should we refactor them into a single hook `hooks/useMdComponents.ts` so the `mdComponents` are consistent between Banner and InlineMessage?

**Review by karrui — COMMENTED** [review]
one last change :P

File: frontend/src/components/InlineMessage/InlineMessage.tsx (hunk 9 above)
Comment (tshuli) [reply frontend/src/components/InlineMessage/InlineMessage.tsx]: added, thanks for going through with me!

File: frontend/src/components/Banner/Banner.tsx [hunk 10]
```diff
@@ -16,7 +14,7 @@ import { BxsErrorCircle, BxsInfoCircle } from '~/assets/icons'
 
 import { BannerVariant } from '~theme/components/Banner'
 
-import Link from '../Link'
+import { useMdComponents } from '../../hooks/useMdComponents'
```
Comment (karrui) [inline frontend/src/components/Banner/Banner.tsx:17]: nit: should we use the alias path instead?

```suggestion
import { useMdComponents } from '~hooks/useMdComponents'
```

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 11]
```diff
@@ -1,12 +1,10 @@
-import { useMemo } from 'react'
 import ReactMarkdown from 'react-markdown'
-import { Components } from 'react-markdown/src/ast-to-react'
 import { Box, Flex, Icon, useMultiStyleConfig } from '@chakra-ui/react'
 
 import { BxsErrorCircle, BxsInfoCircle } from '~/assets/icons'
 import { InlineMessageVariant } from '~/theme/components/InlineMessage'
 
-import Link from '../Link'
+import { useMdComponents } from '../../hooks/useMdComponents'
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:7]: nit: should we use the alias path instead?

```suggestion
import { useMdComponents } from '~hooks/useMdComponents'
```

**Review by karrui — APPROVED** [review]
lgtm with nits

File: frontend/src/components/InlineMessage/InlineMessage.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  variant = 'info',
+  children,
+  useMarkdown = false,
+}: InlineMessageProps): JSX.Element => {
+  const styles = useMultiStyleConfig('InlineMessage', { variant })
+
+  const mdComponents = useMdComponents(styles)
+
+  return (
+    <Box __css={styles.messagebox}>
```
Comment (karrui) [inline frontend/src/components/InlineMessage/InlineMessage.tsx:25]: is there a reason for this outer box?

File: frontend/src/theme/components/InlineMessage.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+export const InlineMessage: ComponentMultiStyleConfig = {
+  parts: ['messagebox', 'item', 'icon'],
+  baseStyle: {
+    messagebox: {
+      padding: '8px',
+    },
+    item: {
+      display: 'flex',
+      py: ['1rem', '1rem', '0.5rem'],
+      px: '1rem',
```
Comment (karrui) [inline frontend/src/theme/components/InlineMessage.ts:14]: The padding seems to be constant at 1rem, don't think there's a difference between breakpoints

File: frontend/src/components/InlineMessage/InlineMessage.tsx (hunk 12 above)
Comment (karrui) [reply frontend/src/components/InlineMessage/InlineMessage.tsx:25]: I think messagebox and item styles can be combined into a single `Box` component

Comment (tshuli) [reply frontend/src/components/InlineMessage/InlineMessage.tsx]: thanks! combined the two

**Pull request merged**